content/blog/e2e-encryption/index.md

```diff
@@ -0,0 +1,65 @@
+---
+path: '/e2e-encryption'
+title: 'Adding End-to-End Encryption for Proxied Data'
+date: 2021-9-21T06:00:00.000+00:00
+featured_image: e2e-hero.png
+categories: ['Pixie Team Blogs']
+authors: ['Vihang Mehta']
+emails: ['vihang@pixielabs.ai']
+featured: true
+---
+
+End-to-end encryption has become increasingly popular as consumers demand that any data they send - a file, email, or text message - is not decipherable by any unauthorized recipients. This consumer trend is evident in the recent surge in popularity of Signal, an encrypted instant messaging service. 
+
+In this post, we’ll cover what end-to-end encryption is and walk you through how we implemented it in our system.
+
+## Why End-to-End Encryption? 
+
+::: div image-xl
+<svg title="This is a simplified architecture diagram of our system before end-to-end encryption." src='before-e2e.svg' />
+:::
+
+We use standard security practices to secure data in transit; all network communication between the cluster, proxy and client is TLS encrypted. 
+
+But TLS encryption is only point-to-point. When data passes internally through our proxy, the data is temporarily unencrypted. Pixie is an open source project, so users might deploy Pixie Cloud (and the accompanying proxy) in a variety of environments. We wanted to provide privacy guarantees for users given the heterogeneity of deployment scenarios. 
+
+By adding end-to-end encryption, we can ensure that the proxy only sees an encrypted form of the telemetry data.
+
+## Implementation
+
+Pixie provides multiple clients for developers to interact with its platform:
+* a web-UI (JavaScript)
+* a CLI (Golang)
+* APIs (client libraries: Golang, Python)
+
+Since we needed to support E2E encryption across multiple languages, using a crypto standard with readily available implementations in multiple languages was a must. Given that we already use JWTs for user claims, we chose to look at the IETF proposed JOSE standard for our E2E encryption needs. We settled on using JWKs for key exchange and JWEs as our encryption format. 
+
+There are multiple libraries that implement the JOSE spec in different languages. We chose the following:
+* [jose](https://www.npmjs.com/package/jose) for JavaScript (imported as [@inrupt/jose-legacy-modules](https://www.npmjs.com/package/@inrupt/jose-legacy-modules) for compatibility with our tooling)
+* [lestrrat-go/jwx](https://pkg.go.dev/github.com/lestrrat-go/jwx) for Golang
+* [Authlib](https://pypi.org/project/Authlib/) for Python (notably, this library successfully handles messages that include null bytes)
+
+## End-to-End Encryption in Pixie
+
+::: div image-xl
```

Comment:
i would add a preamble sentence before the diagram as well. You could reference the above libraries so that it connects to the previous sentence and isn't redundant with the diagram description